migration-engine/connectors/migration-connector/src/steps.rs

```diff
@@ -214,38 +214,37 @@ impl DirectiveLocation {
 }
 
 #[derive(Debug, Deserialize, Serialize, PartialEq, Clone)]
-#[serde(rename_all = "camelCase", deny_unknown_fields, untagged)]
-pub enum DirectiveType {
+#[serde(deny_unknown_fields, tag = "argumentType")]
+pub enum ArgumentType {
     TypeAlias { type_alias: String },
-    Field { model: String, field: String },
-    Model { model: String },
-    Enum { r#enum: String },
+    FieldDirective { model: String, field: String },
+    ModelDirective { model: String },
+    EnumDirective { r#enum: String },
+    Datasource,
+    Generator,
 }
```

Comment:
Regarding the `ArgumentType`/`ArgumentLocation` duo, we already talked about it briefly when pairing but I was thinking another shape could make sense:

```rust
// The current ArgumentType, basically return it to its previous state.
pub enum DirectiveType {
    Field { model: String, field: String },
    Model { model: String },
    Enum { r#enum: String },
}

pub enum ArgumentLocation {
    Directive { directive_type: DirectiveType, directive: String, arguments: Option<Vec<Argument>> },
    Source { source: String },
   // and maybe Generator
}
```

That has two potential upsides:

- I think we don't need the two `unreachable`s anymore
- We can go back to flat JSON objects instead of a nested `location` object. But as we discussed I'm not sure flat is better, the nested `location` as it is now is more explicit, so maybe better. We could also nest with ArgumentLocation as an enum though.